<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        <meta name="author" content="Yang Fang">
        
        <link rel="shortcut icon" href="../img/favicon.ico">

	<title>Tutorial - Physpetools</title>

        <link href="../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../css/font-awesome-4.5.0.css" rel="stylesheet">
        <link href="../css/base.css" rel="stylesheet">
        <link rel="stylesheet" href="../css/highlight.css">

        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

        
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            

            <!-- Main title -->
            <a class="navbar-brand" href="..">Physpetools</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
            
                <!-- Main navigation -->
                <ul class="nav navbar-nav">
                
                
                    <li >
                        <a href="..">Home</a>
                    </li>
                
                
                
                    <li >
                        <a href="../install/">Install</a>
                    </li>
                
                
                
                    <li class="dropdown active">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Documents <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../usage/">Usage</a>
</li>

                        
                            
<li class="active">
    <a href="./">Tutorial</a>
</li>

                        
                        </ul>
                    </li>
                
                
                
                    <li >
                        <a href="../faq/">FAQ</a>
                    </li>
                
                
                </ul>
            

            <ul class="nav navbar-nav navbar-right">
                <li>
                    <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                        <i class="fa fa-search"></i> Search
                    </a>
                </li>
                
                    <li >
                        <a rel="next" href="../usage/">
                            <i class="fa fa-arrow-left"></i> Previous
                        </a>
                    </li>
                    <li >
                        <a rel="prev" href="../faq/">
                            Next <i class="fa fa-arrow-right"></i>
                        </a>
                    </li>
                
                
                    <li>
                        <a href="https://github.com/xiaofeiyangyang/physpetools">
                            
                                <i class="fa fa-github"></i>
                            
                            GitHub
                        </a>
                    </li>
                
            </ul>
        </div>
    </div>
</div>

        <div class="container">
            
                <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
    
        <li class="main active"><a href="#physpetree-tutorial">PhySpeTree Tutorial</a></li>
        
            <li><a href="#automatically-reconstruct-tree-of-life">Automatically reconstruct Tree of Life</a></li>
        
            <li><a href="#auto-build-a-plant-tree">Auto build a plant tree</a></li>
        
            <li><a href="#extend-tree-with-a-new-organism">Extend tree with a new organism</a></li>
        
    
    </ul>
</div></div>
                <div class="col-md-9" role="main">

<h1 id="physpetree-tutorial">PhySpeTree Tutorial</h1>
<h2 id="automatically-reconstruct-tree-of-life">Automatically reconstruct Tree of Life</h2>
<p>The Tree of Life always used to describe the relations between organisms are living and extinct.
Here we reconstruct tree of life, which contain 190 organisms. We use two method automatically reconstruct phylogenetic tree. One way is use highly
conserved protein method and another is use SSU rRNA method to reconstruct Tree of Life.</p>
<h3 id="auto-build-tree-of-life-by-highly-conserved-protein-method">Auto build Tree of life by highly conserved protein method</h3>
<ul>
<li><strong>Concept</strong></li>
</ul>
<p><img alt="physpe_concept_hcp" src="../img/physpe_concept_hcp.png" /></p>
<p>In highly conserved protein (hcp) method automatically build phylogenetic tree by concatenate highly conserved protein to a supermatrix then build tree.
Here we prepare 31 highly conserved proteins to automatically reconstruct phylogenetic tree. you also can searching what's the <a href="../faq/#Physpe_reconstruct_phylogenetic_tree_database">31 highly protein</a> be used.</p>
<h4 id="1prepare-species-names-list">1.Prepare species names list</h4>
<p>Preparing the organisms names (abbreviated names) are same with <a href="http://www.genome.jp/kegg/catalog/org_list.html">KEGG organisms</a> abbreviation. you can get KEGG organisms abbreviation form <a href="http://rest.kegg.jp/list/organism">KEGG API</a>. 
Here we reconstructed tree of life use 191 organisms names are <a href="../example/191speciesnames.txt">download</a>. the organisms names list as follow:</p>
<pre><code class="bash">$ cat 191speciesnames.txt 
neq
pai
ape
sto
ssoa
tvo
tac
afu
.....
</code></pre>

<h4 id="2reconstruct-phylogenetic-tree-by-highly-conserved-protein-method">2.Reconstruct phylogenetic tree by highly conserved protein method</h4>
<p>When users use PhySpeTree automatically build phylogenetic tree, the users can use <code>--hcp</code> arguments to specify use highly conserved protein method to build tree. The default use 1 thread 
to reconstruct phylogenetic tree. If you want to use more threads can use <code>-t</code> parameter to set the threads to be used. Here we used the default value 1 thread to 
build 191 species phylogenetic tree.</p>
<pre><code>$ PhySpeTree autobuild -i 191speciesnames.txt -o 191_pro --hcp
Loading organisms names success.....

The result are store in:191_pro

Now loading data and constructing species phylogenetic tree......
2016-09-06 13:01:53,462 Checking organisms INFO: The organism: ges
2016-09-06 13:01:53,462 Checking organisms WARNING: There organisms can't match in KEGG database so removed and reconstruct phylogenetic tree
2016-09-06 13:01:53,462 kegg DB INFO: Read organisms names success
2016-09-06 13:02:15,058 kegg DB INFO: Retrieve highly conserved protein 'Ribosomal protein L1' success and store in p1.fasta file
2016-09-06 13:02:37,771 kegg DB INFO: Retrieve highly conserved protein 'Leucyl-tRNA synthetase' success and store in p2.fasta file
2016-09-06 13:03:00,211 kegg DB INFO: Retrieve highly conserved protein 'Ribosomal protein L14' success and store in p3.fasta file
2016-09-06 13:03:20,662 kegg DB INFO: Retrieve highly conserved protein 'Ribosomal protein L5' success and store in p4.fasta file
2016-09-06 13:03:40,959 kegg DB INFO: Retrieve highly conserved protein 'Ribosomal protein S7' success and store in p5.fasta file
2016-09-06 13:04:01,849 kegg DB INFO: Retrieve highly conserved protein 'Ribosomal protein S8' success and store in p6.fasta file
2016-09-06 13:04:24,221 kegg DB INFO: Retrieve highly conserved protein 'Arginyl-tRNA synthetase' success and store in p7.fasta file
2016-09-06 13:04:24,222 kegg DB INFO: retrieve from Kegg DB 7 highly conserved proteins

</code></pre>

<h4 id="3get-the-tree-file">3.Get the tree file</h4>
<p><img alt="191_species_rna" src="../img/191_species_pro.png" /></p>
<h4 id="4use-iview-module-annotating-tree">4.Use iview module annotating tree</h4>
<p>PhySpeTree provide iview module to annotating tree by iTol, iTol is a very popular online tool for the display, annotation and management of phylogenetic trees.
When you use iview module to annotating tree, only drop the output the generate files to your iTol account to display tree.</p>
<ul>
<li>Change abbreviated label names to full names use <code>--labels</code> arguments.</li>
</ul>
<pre><code>$ PhySpeTree iview -i 191speciesnames.txt --labels
Change abbreviation names to full names complete
change labels file was save in iview/labels.txt

$ cd iview

$ cat labels.txt 
LABELS
SEPARATOR TAB
DATA
neq     Nanoarchaeum equitans
pai     Pyrobaculum aerophilum
ape     Aeropyrum pernix
sto     Sulfolobus tokodaii
ssoa    Sulfolobus solfataricus SULA
tvo     Thermoplasma volcanium
tac     Thermoplasma acidophilum
afu     Archaeoglobus fulgidus DSM 4304
hal     Halobacterium sp. NRC-1
mac     Methanosarcina acetivorans
mma     Methanosarcina mazei Go1
pfu     Pyrococcus furiosus DSM 3638
pho     Pyrococcus horikoshii
pab     Pyrococcus abyssi
mth     Methanothermobacter thermautotrophicus
mka     Methanopyrus kandleri
mmp     Methanococcus maripaludis S2
.....
</code></pre>

<ul>
<li>Drop <code>labels.txt</code> to you iTol account and view the phylogenetic tree.</li>
</ul>
<p><img alt="191_species_rna_full_name" src="../img/191_species_pro_full_name.png" /></p>
<ul>
<li>Color range by phylum use <code>-a</code> parameter to specified. The default annotation by phylum.</li>
</ul>
<pre><code>$ PhySpeTree iview -i 191speciesnames.txt -o iview --range -a phylum
Color range by phylum was complete.
Color range annotation was save in iview/range_color_by_phylum.txt

$ cd iview

$ cat range_color_by_phylum.txt 
TREE_COLORS
SEPARATOR TAB
DATA
neq     range   #996433 Archaea
pai     range   #996433 Archaea
ape     range   #996433 Archaea
sto     range   #996433 Archaea
ssoa    range   #996433 Archaea
tvo     range   #996433 Archaea
tac     range   #996433 Archaea
afu     range   #996433 Archaea
hal     range   #996433 Archaea
mac     range   #996433 Archaea
mma     range   #996433 Archaea
pfu     range   #996433 Archaea
pho     range   #996433 Archaea
pab     range   #996433 Archaea
mth     range   #996433 Archaea
mka     range   #996433 Archaea
mmp     range   #996433 Archaea
mja     range   #996433 Archaea
.....
</code></pre>

<ul>
<li>Drop <code>range_color_by_phylum.txt</code> to you iTol account and view the phylogenetic tree.</li>
</ul>
<p><img alt="191_species_rna_full_color" src="../img/191_species_pro_full_name_color.png" /></p>
<h3 id="auto-build-tree-of-life-by-16s-ssu-rna">Auto build Tree of Life by 16s SSU RNA</h3>
<ul>
<li><strong>Concept</strong></li>
</ul>
<p><img alt="physpe_concept_hcp" src="../img/physpe_concept_rna.png" /></p>
<p>PhySpeTree provide auto reconstruct phylogenetic tree by 16s SSU RNA, the concept of this way is use alignment organisms 16s SSU RNA and then reconstruct 
phylogenetic tree.</p>
<h4 id="1prepare-organism-names-list-191-organisms-names-list-download">1.Prepare organism names list, 191 organisms names list <a href="../example/191speciesnames.txt">download</a></h4>
<pre><code class="bash">$ cat 191speciesnames.txt 
neq
pai
ape
sto
ssoa
tvo
tac
afu
.....
</code></pre>

<h4 id="2reconstruct-phylogenetic-tree-by-16s-ssu-rna-method">2.Reconstruct phylogenetic tree by 16s SSU RNA method</h4>
<pre><code>$ PhySpeTree autobuild -i 191speciesnames.txt -o 191_rna --srna
Loading organisms names success.....

The result are store in:191_rna

Now loading data and constructing species phylogenetic tree......
2016-09-05 19:48:03,824 Checking organisms INFO: The organism: neq
2016-09-05 19:48:03,824 Checking organisms INFO: The organism: ape
2016-09-05 19:48:03,824 Checking organisms INFO: The organism: tac
2016-09-05 19:48:03,824 Checking organisms INFO: The organism: mmp
2016-09-05 19:48:03,824 Checking organisms INFO: The organism: gla
2016-09-05 19:48:03,824 Checking organisms INFO: The organism: tps
2016-09-05 19:48:03,824 Checking organisms INFO: The organism: cho
2016-09-05 19:48:03,824 Checking organisms INFO: The organism: ddi
2016-09-05 19:48:03,824 Checking organisms INFO: The organism: spo
2016-09-05 19:48:03,824 Checking organisms INFO: The organism: aga
2016-09-05 19:48:03,824 Checking organisms INFO: The organism: tru
2016-09-05 19:48:03,824 Checking organisms INFO: The organism: mpu
2016-09-05 19:48:03,824 Checking organisms INFO: The organism: lin
2016-09-05 19:48:03,825 Checking organisms INFO: The organism: ban
2016-09-05 19:48:03,825 Checking organisms INFO: The organism: bce
2016-09-05 19:48:03,825 Checking organisms INFO: The organism: ljo
2016-09-05 19:48:03,825 Checking organisms INFO: The organism: san
2016-09-05 19:48:03,825 Checking organisms INFO: The organism: spg
2016-09-05 19:48:03,825 Checking organisms INFO: The organism: ges
2016-09-05 19:48:03,825 Checking organisms INFO: The organism: lis
2016-09-05 19:48:03,825 Checking organisms INFO: The organism: sco
2016-09-05 19:48:03,825 Checking organisms INFO: The organism: cdi
2016-09-05 19:48:03,825 Checking organisms INFO: The organism: mle
2016-09-05 19:48:03,825 Checking organisms INFO: The organism: wsu
2016-09-05 19:48:03,825 Checking organisms INFO: The organism: rpr
2016-09-05 19:48:03,825 Checking organisms INFO: The organism: bpe
2016-09-05 19:48:03,825 Checking organisms INFO: The organism: bpa
2016-09-05 19:48:03,825 Checking organisms INFO: The organism: ppr
2016-09-05 19:48:03,825 Checking organisms WARNING: There organisms can't match in SILVA database so removed and reconstruct phylogenetic tree
2016-09-05 19:48:03,825 16s DB INFO: Read organisms names success
2016-09-05 19:48:14,354 16s DB INFO: Retrieve organism 'pai' 16s SSU RNA sequences data success
2016-09-05 19:48:17,703 16s DB INFO: Retrieve organism 'sto' 16s SSU RNA sequences data success
2016-09-05 19:48:20,674 16s DB INFO: Retrieve organism 'ssoa' 16s SSU RNA sequences data success
2016-09-05 19:48:23,031 16s DB INFO: Retrieve organism 'tvo' 16s SSU RNA sequences data success
2016-09-05 19:48:26,510 16s DB INFO: Retrieve organism 'afu' 16s SSU RNA sequences data success
2016-09-05 19:48:28,766 16s DB INFO: Retrieve organism 'hal' 16s SSU RNA sequences data success
2016-09-05 19:48:31,148 16s DB INFO: Retrieve organism 'mac' 16s SSU RNA sequences data success
2016-09-05 19:48:33,579 16s DB INFO: Retrieve organism 'mma' 16s SSU RNA sequences data success
2016-09-05 19:48:35,856 16s DB INFO: Retrieve organism 'pfu' 16s SSU RNA sequences data success
2016-09-05 19:48:38,194 16s DB INFO: Retrieve organism 'pho' 16s SSU RNA sequences data success
......
</code></pre>

<h4 id="3get-the-tree-file_1">3.Get the tree file</h4>
<p><img alt="191_species_rna" src="../img/191_species_rna.png" /></p>
<h4 id="4use-iview-annotating-tree">4.Use iview annotating tree</h4>
<ul>
<li>Change abbreviated label names to full names use <code>--labels</code></li>
</ul>
<pre><code>$ PhySpeTree iview -i 191speciesnames.txt --labels
Change abbreviation names to full names complete
change labels file was save in iview/labels.txt

$ cd iview

$ cat labels.txt 
LABELS
SEPARATOR TAB
DATA
neq     Nanoarchaeum equitans
pai     Pyrobaculum aerophilum
ape     Aeropyrum pernix
sto     Sulfolobus tokodaii
ssoa    Sulfolobus solfataricus SULA
tvo     Thermoplasma volcanium
tac     Thermoplasma acidophilum
afu     Archaeoglobus fulgidus DSM 4304
hal     Halobacterium sp. NRC-1
mac     Methanosarcina acetivorans
mma     Methanosarcina mazei Go1
pfu     Pyrococcus furiosus DSM 3638
pho     Pyrococcus horikoshii
pab     Pyrococcus abyssi
mth     Methanothermobacter thermautotrophicus
mka     Methanopyrus kandleri
mmp     Methanococcus maripaludis S2
.....
</code></pre>

<ul>
<li>Tree view in iTol</li>
</ul>
<p><img alt="191_species_rna_full_name" src="../img/191_species_rna_full_name.png" /></p>
<ul>
<li>Color range by phylum</li>
</ul>
<pre><code>$ PhySpeTree iview -i 191speciesnames.txt -o iview --range -a phylum
Color range by phylum was complete.
Color range annotation was save in iview/range_color_by_phylum.txt

$ cd iview

$ cat range_color_by_phylum.txt 
TREE_COLORS
SEPARATOR TAB
DATA
neq     range   #996433 Archaea
pai     range   #996433 Archaea
ape     range   #996433 Archaea
sto     range   #996433 Archaea
ssoa    range   #996433 Archaea
tvo     range   #996433 Archaea
tac     range   #996433 Archaea
afu     range   #996433 Archaea
hal     range   #996433 Archaea
mac     range   #996433 Archaea
mma     range   #996433 Archaea
pfu     range   #996433 Archaea
pho     range   #996433 Archaea
pab     range   #996433 Archaea
mth     range   #996433 Archaea
mka     range   #996433 Archaea
mmp     range   #996433 Archaea
mja     range   #996433 Archaea
.....
</code></pre>

<ul>
<li>Tree view in iTol</li>
</ul>
<p><img alt="191_species_rna_full_color" src="../img/191_species_rna_full_name_color.png" /></p>
<ul>
<li>Color range by class</li>
</ul>
<pre><code>$ PhySpeTree iview -i 191speciesnames.txt --range -a class
Color range by class was complete.
Color range annotation was save in iview/range_color_by_class.txt

$ cd iview

$ cat range_color_by_class.txt 
TREE_COLORS
SEPARATOR TAB
DATA
neq     range   #4A959E Nanoarchaeota
pai     range   #58CD80 Crenarchaeota
ape     range   #58CD80 Crenarchaeota
sto     range   #58CD80 Crenarchaeota
ssoa    range   #58CD80 Crenarchaeota
tvo     range   #639BB0 Euryarchaeota
tac     range   #639BB0 Euryarchaeota
afu     range   #639BB0 Euryarchaeota
hal     range   #639BB0 Euryarchaeota
mac     range   #639BB0 Euryarchaeota
mma     range   #639BB0 Euryarchaeota
pfu     range   #639BB0 Euryarchaeota
pho     range   #639BB0 Euryarchaeota
pab     range   #639BB0 Euryarchaeota
mth     range   #639BB0 Euryarchaeota
mka     range   #639BB0 Euryarchaeota
mmp     range   #639BB0 Euryarchaeota
mja     range   #639BB0 Euryarchaeota
gla     range   #C5D49E Diplomonads
lma     range   #899DDB Euglenozoa
tps     range   #7DD2ED Stramenopiles
cho     range   #99A01A Alveolates
.....
</code></pre>

<ul>
<li>Tree view in iTol</li>
</ul>
<p><img alt="191_species_rna_full_color_class" src="../img/191_species_rna_full_name_color_class.png" /></p>
<h2 id="auto-build-a-plant-tree">Auto build a plant tree</h2>
<h4 id="1prepare-organisms-names-list">1.Prepare organisms names list</h4>
<p>Here we automatically reconstructed 52 plants phylogenetic tree by PhySpeTree use highly conserved protein method. The species list <a href="../example/52plantsnames.txt">download</a>.</p>
<pre><code>$ cat 52plantsnames.txt
aly
ath
atr
bdi
bpg
brp
bvg
cam
ccp
cic
cit
cme
cmo
.....
</code></pre>

<h4 id="2reconstruct-phylogenetic-tree-by-highly-conserved-protein-method_1">2.Reconstruct phylogenetic tree by highly conserved protein method</h4>
<pre><code>$ PhySpeTree autobuild -i 52plantsnames.txt -o 52plant_pro --srna -t 6
Loading organisms names success.....

The result are store in:52plant_pro

Now loading data and constructing species phylogenetic tree......
Wed, 07 Sep 2016 18:48:35 kegg DB[line:105] INFO Read organisms names success
Wed, 07 Sep 2016 18:48:44 kegg DB[line:92] INFO Retrieve highly conserved protein 'Leucyl-tRNA synthetase' success and store in p1.fasta file
Wed, 07 Sep 2016 18:48:53 kegg DB[line:92] INFO Retrieve highly conserved protein 'Metal-dependent proteases with chaperone activity' success and store in p2.fasta file
Wed, 07 Sep 2016 18:49:01 kegg DB[line:92] INFO Retrieve highly conserved protein 'Phenylalanine-tRNA synthethase alpha subunit' success and store in p3.fasta file
Wed, 07 Sep 2016 18:49:10 kegg DB[line:92] INFO Retrieve highly conserved protein 'Preprotein translocase subunit SecY' success and store in p4.fasta file
Wed, 07 Sep 2016 18:49:17 kegg DB[line:92] INFO Retrieve highly conserved protein 'Ribosomal protein L11' success and store in p5.fasta file
Wed, 07 Sep 2016 18:49:25 kegg DB[line:92] INFO Retrieve highly conserved protein 'Ribosomal protein L13' success and store in p6.fasta file
Wed, 07 Sep 2016 18:49:32 kegg DB[line:92] INFO Retrieve highly conserved protein 'Ribosomal protein L15' success and store in p7.fasta file
Wed, 07 Sep 2016 18:49:40 kegg DB[line:92] INFO Retrieve highly conserved protein 'Ribosomal protein L16/L10E' success and store in p8.fasta file
Wed, 07 Sep 2016 18:49:48 kegg DB[line:92] INFO Retrieve highly conserved protein 'Ribosomal protein L18' success and store in p9.fasta file
Wed, 07 Sep 2016 18:50:01 kegg DB[line:92] INFO Retrieve highly conserved protein 'Ribosomal protein L22' success and store in p10.fasta file
Wed, 07 Sep 2016 18:50:08 kegg DB[line:92] INFO Retrieve highly conserved protein 'Ribosomal protein L3' success and store in p11.fasta file
Wed, 07 Sep 2016 18:50:16 kegg DB[line:92] INFO Retrieve highly conserved protein 'Ribosomal protein L5' success and store in p12.fasta file
Wed, 07 Sep 2016 18:50:24 kegg DB[line:92] INFO Retrieve highly conserved protein 'Ribosomal protein L6P/L9E' success and store in p13.fasta file
Wed, 07 Sep 2016 18:50:31 kegg DB[line:92] INFO Retrieve highly conserved protein 'Ribosomal protein S11' success and store in p14.fasta file
Wed, 07 Sep 2016 18:50:40 kegg DB[line:92] INFO Retrieve highly conserved protein 'Ribosomal protein S17' success and store in p15.fasta file
Wed, 07 Sep 2016 18:50:48 kegg DB[line:92] INFO Retrieve highly conserved protein 'Ribosomal protein S2' success and store in p16.fasta file
Wed, 07 Sep 2016 18:50:56 kegg DB[line:92] INFO Retrieve highly conserved protein 'Ribosomal protein S4' success and store in p17.fasta file
Wed, 07 Sep 2016 18:51:04 kegg DB[line:92] INFO Retrieve highly conserved protein 'Ribosomal protein S5' success and store in p18.fasta file
Wed, 07 Sep 2016 18:51:11 kegg DB[line:92] INFO Retrieve highly conserved protein 'Ribosomal protein S8' success and store in p19.fasta file
Wed, 07 Sep 2016 18:51:25 kegg DB[line:92] INFO Retrieve highly conserved protein 'Seryl-tRNA synthetase' success and store in p20.fasta file
Wed, 07 Sep 2016 18:51:39 kegg DB[line:92] INFO Retrieve highly conserved protein 'Ribosomal protein S13' success and store in p21.fasta file
Wed, 07 Sep 2016 18:51:39 kegg DB[line:94] INFO retrieve from Kegg DB 21 highly conserved proteins
......
</code></pre>

<h4 id="3get-tree-file">3.Get tree file</h4>
<p><img alt="52_plants_pro" src="../img/52_plants_pro.png" /></p>
<h4 id="4use-iview-annotating-tree_1">4.Use iview annotating tree</h4>
<ul>
<li>Change label names use <code>--labels</code></li>
</ul>
<pre><code>$ PhySpeTree iview -i 52plantsnames.txt --labels
Change abbreviation names to full names complete
change labels file was save in iview/labels.txt

$ cd iview

$ cat labels.txt 
LABELS
SEPARATOR TAB
DATA
aly     Arabidopsis lyrata (lyrate rockcress)
ath     Arabidopsis thaliana (thale cress)
atr     Amborella trichopoda
bdi     Brachypodium distachyon
bpg     Bathycoccus prasinos
brp     Brassica rapa (field mustard)
bvg     Beta vulgaris (sugar beet)
cam     Cicer arietinum (chickpea)
ccp     Chondrus crispus (carragheen)
cic     Citrus clementina (mandarin orange)
cit     Citrus sinensis (Valencia orange)
cme     Cyanidioschyzon merolae
cmo     Cucumis melo (muskmelon)
crb     Capsella rubella
cre     Chlamydomonas reinhardtii
csl     Coccomyxa subellipsoidea
csv     Cucumis sativus (cucumber)
cvr     Chlorella variabilis

.....
</code></pre>

<ul>
<li>Tree view by iTol</li>
</ul>
<p><img alt="52_plants_pro_full_name" src="../img/52_plants_pro_full_name.png" /></p>
<ul>
<li>Color range by class</li>
</ul>
<pre><code>$ PhySpeTree iview -i 52plantsnames.txt -o iview --range -a phylum
Color range by class was complete.
Color range annotation was save in iview/range_color_by_class.txt
$ cd iview

$ cat range_color_by_class.txt 
TREE_COLORS
SEPARATOR TAB
DATA
aly     range   #1C748D Eudicots
ath     range   #1C748D Eudicots
atr     range   #C4DE94 Basal Magnoliophyta
bdi     range   #854B9F Monocots
bpg     range   #784CE5 Green algae
brp     range   #1C748D Eudicots
bvg     range   #1C748D Eudicots
cam     range   #1C748D Eudicots
ccp     range   #CEDDB3 Red algae
cic     range   #1C748D Eudicots
cit     range   #1C748D Eudicots
cme     range   #CEDDB3 Red algae
cmo     range   #1C748D Eudicots
crb     range   #1C748D Eudicots
cre     range   #784CE5 Green algae
csl     range   #784CE5 Green algae
.....
</code></pre>

<ul>
<li>Tree view use iview module by iTol</li>
</ul>
<p><img alt="52_plants_pro_full_name_color" src="../img/52_plants_pro_full_name_color.png" /></p>
<h2 id="extend-tree-with-a-new-organism">Extend tree with a new organism</h2>
<p>Some time we want to extend tree with a new organisms, use PhySpeTree user can easy do it.</p>
<h3 id="extend-tree-by-16s-ssu-rrna-method">Extend tree by 16s SSU rRNA method</h3>
<ul>
<li><strong>Concept</strong></li>
</ul>
<p><img alt="physpe_concept_erna" src="../img/physpe_concept_erna.png" /></p>
<p>Here we extend tree of life with a new organism <code>Lokiarchaeum sp. GC14_75 (loki)</code>.</p>
<h4 id="1prepare-new-organism-ssu-rrna-sequence">1.Prepare new organism SSU rRNA sequence</h4>
<p>When use SSU rRNA method to extend a new organism user need prepare the new organism SSU rRNA sequence, Here we prepared the new organism <code>Lokiarchaeum sp. GC14_75 (loki)</code>
16s rRNA sequence save with FASTA format. example download <a href="../example/extend_rna_loki.fasta">extend_rna_olki.fasta</a></p>
<pre><code>$ cat extend_rna_olki.fasta
&gt;loki
GAGAUGGGUACUGAGACAACGACCCAGGCCUUACGAGGCGCAGCAGGCGCGAAACCUCCGCAAUACACGAAAGUGUGACG
GGGUUACCCAAAGUGUUCAAUUAUGAACUGUGGUAGGUGAGUAAUGUUCCCUACUAGAAAGGAGAGGGCAAGGCUGGUGC
CAGCCGCCGCGGUAAAACCAGCUCUUCAAGUGGUCGGGAUAAUUAUUGGGCUUAAAGUGUCCGUAGCCGGUUUAGUAAGU
UCCUGGUAAAAUCGGGUAGCUUAACUAUCUGUAUGCUAGGAAUACUGCUAUACUAGAGGACGGGAGAGGUCUGAGGUACU
ACAGGGGUAGGGGUGAAAUCUUAUAAUCCUUGUAGGACCACCAGUGGCGAAGGCGUCAGACUGGAACGUGCCUGACGGUG
AGGGACGAAAGCCAGGGGAGCGAACCGGAUUAGAUACCCGGGUAGUCCUGGCCGUAAACGAUGCAUACUAGGUGAUGGCA
UGGCCAUGAGCCAUGUCAGUGCCGUAGGGAAACCGUUAAGUGUGCCGCCUGGGAAGUACGGUCGCAAGGCUAAAACUUAA
AGGAAUUGGCGGGGGAGCACCACAAGGGGUGAAGCCUGCGGUUCAAUUGGACUCAACGCCGGGAAACUUACCAGGGGAGA
CAGCAGAAUGAUGGUCAGGUUGACGACCUUACCUGACAAGCUGAGAGGAGGUGCAUGGCCGUCGCCAGUUCGUGCUGUGA
GGUAUCCUGUUAAGUCAGGCAACGAACGAGAUCCGCACCUUUAUUUGCCAGCAAGAAGUCACGACUUCGUUGGGAACACU
AAAGGGACCGCCGUCGAUAAGACGGAGGAAGGAGCGGGCAAAGGCAGGUCAGUAUGCCCCGAAACCCCUGGGCUACACGC
GGGCUGCAAUGGUAUGAACAAUGGGCUGUAACUCCGAAAGGAGAAACCAAUCCCGAAAUCAUAUCUCAGUGGGAAUUGUC
GGCUGUAACCCGCCGACAUGAACGUGGAAUCCCUAGUAAUCGUGUGUCAUCAUCGCACGGUGAAUACGUCUCUGCUCCUU
GCACACACCGCCCGUCGCUCCAUCCGAGUGUGCUAAAAAUGAGGUAUGGUCAGUCUGGUCGUAUCGAAUUUCUAGUAUGC
GAGGGGGGAGAAGUCGUAACAAGGUAGCCGUAGGGGAACCUGCGGCUGGAUCACCUCCU

</code></pre>

<h4 id="2reconstruct-phylogenetic-tree-by-ssu-rrna-extend-with-a-new-organism">2.Reconstruct phylogenetic tree by SSU rRNA extend with a new organism</h4>
<p>Reconstruct phylogenetic tree by SSU rRNA method, you should use <code>--esrna</code> to specify the method and use <code>-e</code> parameter specify
the extend organism SSU rRNA sequence file.</p>
<pre><code>$ PhySpeTree autobuild -i 191speciesnames.txt -o extend_rna -e extend_rna_loki.fasta --esrna -t 6
Loading organisms names success.....

The result are store in:extend_rna

Now loading data and constructing species phylogenetic tree......
2016-09-08 14:49:15,423 Checking organisms INFO: The organism: neq
2016-09-08 14:49:15,424 Checking organisms INFO: The organism: ape
2016-09-08 14:49:15,424 Checking organisms INFO: The organism: tac
2016-09-08 14:49:15,424 Checking organisms INFO: The organism: mmp
2016-09-08 14:49:15,424 Checking organisms INFO: The organism: gla
2016-09-08 14:49:15,424 Checking organisms INFO: The organism: tps
2016-09-08 14:49:15,424 Checking organisms INFO: The organism: cho
2016-09-08 14:49:15,424 Checking organisms INFO: The organism: ddi
2016-09-08 14:49:15,424 Checking organisms INFO: The organism: spo
2016-09-08 14:49:15,424 Checking organisms INFO: The organism: aga
2016-09-08 14:49:15,424 Checking organisms INFO: The organism: tru
2016-09-08 14:49:15,424 Checking organisms INFO: The organism: mpu
2016-09-08 14:49:15,425 Checking organisms INFO: The organism: lin
2016-09-08 14:49:15,425 Checking organisms INFO: The organism: ban
2016-09-08 14:49:15,425 Checking organisms INFO: The organism: bce
2016-09-08 14:49:15,425 Checking organisms INFO: The organism: ljo
2016-09-08 14:49:15,425 Checking organisms INFO: The organism: san
2016-09-08 14:49:15,425 Checking organisms INFO: The organism: spg
2016-09-08 14:49:15,425 Checking organisms INFO: The organism: ges
2016-09-08 14:49:15,425 Checking organisms INFO: The organism: lis
2016-09-08 14:49:15,425 Checking organisms INFO: The organism: sco
2016-09-08 14:49:15,425 Checking organisms INFO: The organism: cdi
2016-09-08 14:49:15,425 Checking organisms INFO: The organism: mle
2016-09-08 14:49:15,425 Checking organisms INFO: The organism: wsu
2016-09-08 14:49:15,425 Checking organisms INFO: The organism: rpr
2016-09-08 14:49:15,425 Checking organisms INFO: The organism: bpe
2016-09-08 14:49:15,425 Checking organisms INFO: The organism: bpa
2016-09-08 14:49:15,425 Checking organisms INFO: The organism: ppr
2016-09-08 14:49:15,425 Checking organisms WARNING: There organisms can't match in SILVA database so removed and reconstruct phylogenetic tree
2016-09-08 14:49:15,425 16s DB INFO: Read organisms names success
2016-09-08 14:49:15,499 16s DB INFO: Retrieve organism 'pai' 16s SSU RNA sequences data success
2016-09-08 14:49:15,515 16s DB INFO: Retrieve organism 'sto' 16s SSU RNA sequences data success
2016-09-08 14:49:15,532 16s DB INFO: Retrieve organism 'ssoa' 16s SSU RNA sequences data success
2016-09-08 14:49:15,559 16s DB INFO: Retrieve organism 'tvo' 16s SSU RNA sequences data success
2016-09-08 14:49:15,577 16s DB INFO: Retrieve organism 'afu' 16s SSU RNA sequences data success
.....
</code></pre>

<h4 id="3view-tree-by-itol">3.View tree by iTol</h4>
<p><img alt="extend_rna" src="../img/extend_rna.png" /></p>
<h3 id="extend-tree-by-highly-conserved-protein-hcp-method">Extend tree by highly conserved protein (hcp) method</h3>
<ul>
<li><strong>Concept</strong></li>
</ul>
<p><img alt="physpe_concept_ehcp" src="../img/physpe_concept_ehcp.png" /></p>
<h4 id="1check-prepare-proteins">1.Check prepare proteins</h4>
<p>When use highly conserved protein method reconstruct phylogenetic tree with new organism, user should use <code>check</code> method to check what
highly conserved protein will be prepared.</p>
<pre><code>$ PhySpeTree check -i 191speciesnames.txt --ehcp
'Ribosomal protein L1' ----------------------------------&gt; p1.fasta

'Leucyl-tRNA synthetase' ----------------------------------&gt; p2.fasta

'Ribosomal protein L14' ----------------------------------&gt; p3.fasta

'Ribosomal protein L5' ----------------------------------&gt; p4.fasta

'Ribosomal protein S7' ----------------------------------&gt; p5.fasta

'Ribosomal protein S8' ----------------------------------&gt; p6.fasta

'Arginyl-tRNA synthetase' ----------------------------------&gt; p7.fasta

Check extend highly conserved protein is completed.

Check result is store in /check/physpe_echp_extend.txt
......
</code></pre>

<h4 id="2-prepare-highly-conserved-proteins">2. Prepare highly conserved proteins</h4>
<p>After use check command, we should prepare seven highly conserved protein store this highly conserved protein to p1 ~ p7 FASTA format files.
The highly protein protein names are know with <code>check</code> command. Here we prepare p1 ~ p7 highly conserved proteins to <code>Lokiarchaeum sp. GC14_75 (loki)</code>.
Download <a href="../example/highly_conserved_protein_loki.fasta">highly_conserved_protein_loki</a></p>
<pre><code>$ cd extend_pro_loki
$ ls
p1.fasta  p2.fasta  p3.fasta  p4.fasta  p5.fasta  p6.fasta  p7.fasta

$ cat p1.fasta 
&gt;loki
MKVDDNLLKQSLNAAIDFSVRKKEGFKDRVRKFDETIDLIINIKDVNLNDPKNRIDKEII
LTNEIVEEEKLNICVIASGEILLEAKKAGVETLDRDALIKLNNEEKKHKKKFAKKYEFFI
VEDKMMRDVARYLARFLGPLGKMPKPFPTGYGIISSPGDLRTAVERYKKVIRIQMKKQPI
IFAKIGKKSMEIDRLFDNMKTVIDFIADQMPHKFNNFKSMYLKSSMGKPIKVTEEFLKSL
EV

</code></pre>

<h4 id="3reconstruct-phylogenetic-tree-by-highly-conserved-protein-method-extend-with-a-new-organism">3.Reconstruct phylogenetic tree by highly conserved protein method extend with a new organism</h4>
<pre><code>$ PhySpeTree autobuild -i 191speciesnames.txt -o extend_pro_loki -e highly_consrved_protein_loki --ehcp -t 6
Loading organisms names success.....

The result are store in:extend_pro_loki

Now loading data and constructing species phylogenetic tree......
2016-09-08 15:49:46,391 Checking organisms INFO: The organism: ges
2016-09-08 15:49:46,392 Checking organisms WARNING: There organisms can't match in KEGG database so removed and reconstruct phylogenetic tree
2016-09-08 15:49:46,392 kegg DB INFO: Read organisms names success
2016-09-08 15:50:20,604 kegg DB INFO: Retrieve highly conserved protein 'Ribosomal protein L1' success and store in p1.fasta file
2016-09-08 15:50:57,464 kegg DB INFO: Retrieve highly conserved protein 'Leucyl-tRNA synthetase' success and store in p2.fasta file
2016-09-08 15:51:27,257 kegg DB INFO: Retrieve highly conserved protein 'Ribosomal protein L14' success and store in p3.fasta file
2016-09-08 15:51:55,441 kegg DB INFO: Retrieve highly conserved protein 'Ribosomal protein L5' success and store in p4.fasta file
2016-09-08 15:52:28,429 kegg DB INFO: Retrieve highly conserved protein 'Ribosomal protein S7' success and store in p5.fasta file
2016-09-08 15:52:59,551 kegg DB INFO: Retrieve highly conserved protein 'Ribosomal protein S8' success and store in p6.fasta file
2016-09-08 15:53:31,157 kegg DB INFO: Retrieve highly conserved protein 'Arginyl-tRNA synthetase' success and store in p7.fasta file
2016-09-08 15:53:31,158 kegg DB INFO: retrieve from Kegg DB 7 highly conserved proteins
......

</code></pre>

<h4 id="4view-tree-by-itol">4.View tree by iTol</h4>
<p><img alt="extend_pro" src="../img/extend_pro.png" /></p></div>
            
        </div>

        <footer class="col-md-12">
            <hr>
            
            <p>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</p>
        </footer>

        <script src="../js/jquery-1.10.2.min.js"></script>
        <script src="../js/bootstrap-3.0.3.min.js"></script>
        <script src="../js/highlight.pack.js"></script>
        <script>var base_url = '..';</script>
        <script data-main="../mkdocs/js/search.js" src="../mkdocs/js/require.js"></script>
        <script src="../js/base.js"></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
            <div class="modal-dialog">
                <div class="modal-content">
                    <div class="modal-header">
                        <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                        <h4 class="modal-title" id="exampleModalLabel">Search</h4>
                    </div>
                    <div class="modal-body">
                        <p>
                            From here you can search these documents. Enter
                            your search terms below.
                        </p>
                        <form role="form">
                            <div class="form-group">
                                <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
                            </div>
                        </form>
                        <div id="mkdocs-search-results"></div>
                    </div>
                    <div class="modal-footer">
                    </div>
                </div>
            </div>
        </div>

    </body>
</html>
