<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        <meta name="author" content="Yang Fang">
        
        <link rel="shortcut icon" href="../img/favicon.ico">

	<title>Usage - Physpetools</title>

        <link href="../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../css/font-awesome-4.5.0.css" rel="stylesheet">
        <link href="../css/base.css" rel="stylesheet">
        <link rel="stylesheet" href="../css/highlight.css">

        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

        
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            

            <!-- Main title -->
            <a class="navbar-brand" href="..">Physpetools</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
            
                <!-- Main navigation -->
                <ul class="nav navbar-nav">
                
                
                    <li >
                        <a href="..">Home</a>
                    </li>
                
                
                
                    <li >
                        <a href="../install/">Install</a>
                    </li>
                
                
                
                    <li class="dropdown active">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Documents <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li class="active">
    <a href="./">Usage</a>
</li>

                        
                            
<li >
    <a href="../tutorial/">Tutorial</a>
</li>

                        
                        </ul>
                    </li>
                
                
                
                    <li >
                        <a href="../faq/">FAQ</a>
                    </li>
                
                
                </ul>
            

            <ul class="nav navbar-nav navbar-right">
                <li>
                    <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                        <i class="fa fa-search"></i> Search
                    </a>
                </li>
                
                    <li >
                        <a rel="next" href="../install/">
                            <i class="fa fa-arrow-left"></i> Previous
                        </a>
                    </li>
                    <li >
                        <a rel="prev" href="../tutorial/">
                            Next <i class="fa fa-arrow-right"></i>
                        </a>
                    </li>
                
                
                    <li>
                        <a href="https://github.com/xiaofeiyangyang/physpetools">
                            
                                <i class="fa fa-github"></i>
                            
                            GitHub
                        </a>
                    </li>
                
            </ul>
        </div>
    </div>
</div>

        <div class="container">
            
                <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
    
        <li class="main active"><a href="#usage-and-examples">Usage and Examples</a></li>
        
            <li><a href="#autobuild">autobuild</a></li>
        
            <li><a href="#build">build</a></li>
        
            <li><a href="#combine">combine</a></li>
        
            <li><a href="#iview">iview</a></li>
        
            <li><a href="#check">check</a></li>
        
    
    </ul>
</div></div>
                <div class="col-md-9" role="main">

<h1 id="usage-and-examples">Usage and Examples</h1>
<h2 id="autobuild">autobuild</h2>
<p>User should prepare a txt file contain the abbreviation names of organisms <a href="../example/organism_example_list.txt">organism_example_list.txt</a></p>
<p>Use <strong>autobuild</strong> in command line like this:</p>
<pre><code class="bash">$ physpe -i organism_example_list.txt [options]*
</code></pre>

<h3 id="autobuild-options">Autobuild options</h3>
<table>
<thead>
<tr>
<th align="left">option</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">-h</td>
<td align="left">Print help message and exits.</td>
</tr>
<tr>
<td align="left">-i</td>
<td align="left">Input a txt file contain the a abbreviation species names are same with KEGG species abbreviation.</td>
</tr>
<tr>
<td align="left">-o</td>
<td align="left">A directory include output data (reconstruct tree files). The default output data name is Outdata.</td>
</tr>
<tr>
<td align="left">-t</td>
<td align="left">Specify the number of processing threads (CPUs) to use for Physpe to reconstruct phylogenetic tree. The default is 1.</td>
</tr>
<tr>
<td align="left">-e</td>
<td align="left">The extended data should be FASTA format to extend phylogenetic tree by --ehcp or --esrna option.</td>
</tr>
<tr>
<td align="left">--hcp</td>
<td align="left">The hcp (highly conserved protein) mode is use highly conserved proteins to reconstruct phylogenetic tree. The default mode is hcp.</td>
</tr>
<tr>
<td align="left">--ehcp</td>
<td align="left">The ehcp (extend highly conserved protein) mode is use highly conserved proteins and extend highly protein (user provide) to reconstruct phylogenetic tree.</td>
</tr>
<tr>
<td align="left">--srna</td>
<td align="left">The srna (16s SSU RNA) mode is use 16s SSU RNA data to reconstruct phylogenetic tree.</td>
</tr>
<tr>
<td align="left">--esrna</td>
<td align="left">The esrna (extend 16s SSU RNA) mode is use 16s SSU RNA data and extend 16s SSU RNA (user provide) to reconstruct phylogenetic tree.</td>
</tr>
</tbody>
</table>
<h3 id="example">example</h3>
<p>When use autobuild command to reconstruct phylogenetic tree, user should prepare a organism list with abbreviation names are same with <a href="http://www.genome.jp/kegg/catalog/org_list.html">KEGG organism list</a></p>
<p>List format like follow or download <a href="../example/organism_example_list.txt">organism_example_list.txt</a></p>
<pre><code class="bash">$ cat organism_example_list.txt
aca
ace
acl
acn
aco
acp
adg
adk
aeh
aeq
</code></pre>

<h4 id="auto-build-phylogenetic-tree-by-highly-conserved-proteins"><strong>Auto build phylogenetic tree by highly conserved proteins:</strong></h4>
<pre><code class="bash">$ physpe autobuild -i organism_example_list.txt --hcp
Loading organisms names success.....

The result are store in:Outdata

now loading data and constructing species phylogenetic tree......
Now loading data and constructing species phylogenetic tree......
2016-08-18 15:32:06,770 kegg DB INFO: Read organisms names successful
2016-08-18 15:32:08,380 kegg DB INFO: Retrieve highly conserved protein 'Ribosomal protein L1' success and store in p1.fasta file
2016-08-18 15:32:09,702 kegg DB INFO: Retrieve highly conserved protein 'DNA-directed RNA polymerase subunit alpha' success and store in p2.fasta file
2016-08-18 15:32:11,352 kegg DB INFO: Retrieve highly conserved protein 'Leucyl-tRNA synthetase' success and store in p3.fasta file
2016-08-18 15:32:12,786 kegg DB INFO: Retrieve highly conserved protein 'Metal-dependent proteases with chaperone activity' success and store in p4.fasta file
2016-08-18 15:32:14,318 kegg DB INFO: Retrieve highly conserved protein 'Phenylalanine-tRNA synthethase alpha subunit' success and store in p5.fasta file
2016-08-18 15:32:15,854 kegg DB INFO: Retrieve highly conserved protein 'Predicted GTPase probable translation factor' success and store in p6.fasta file
2016-08-18 15:32:17,049 kegg DB INFO: Retrieve highly conserved protein 'Ribosomal protein L11' success and store in p7.fasta file
2016-08-18 15:32:18,316 kegg DB INFO: Retrieve highly conserved protein 'Ribosomal protein L13' success and store in p8.fasta file

</code></pre>

<p>When auto reconstruct phylogenetic tree by <code>hcp</code> method was completed, you will get file layout like this:</p>
<pre><code>log.log
Outdata/
        RAxML_bestTree.T1
        RAxML_bipartitions.T1
        RAxML_bipartitionsBranchLabels.T1
        RAxML_bootstrap.T1
        RAxML_info.T1
   temp/
        conserved_protein20160817210902/
                                       　p1.fasta
                                       　p2.fasta
                                       　p3.fasta
                                       　......
        hcp_alignment20160817210936/
                                    p1.fasta
                                    p2.fasta
                                    p2.fasta
                                    ......
        concatenate20160817210938/
                                  concatenate.fasta
                                  concatenate.fasta-gb1
                                  concatenate.fasta-gb1.htm
                                  concatenate.fasta-gb1.phy
</code></pre>

<ul>
<li>log.log: The log information of physpe.</li>
<li>
<p>Outdata: Contain phylogenetic tree output result. </p>
<ul>
<li>RAxML_bestTree.T1: Reconstruct phylogenetic tree by RAxML, it's best ML search tree. </li>
<li>RAxML_bipartitions.T1: Bipartition tree by RAxML.</li>
<li>RAxML_bipartitionsBranchLabels.T1: Bipartition tree by constructed by RAxML with branch length lables.</li>
<li>RAxML_bootstrap.T1: Bootstrap result by RAxML</li>
<li>RAxML_info.T1: The info of run RAxMl.</li>
</ul>
</li>
<li>
<p>temp: The temp data by physpe, <code>it's very important to user to check the key steps</code>.</p>
<ul>
<li>conserved_protein: Contain highly conserved proteins retrieve form KEGG database.</li>
<li>hcp_alignment: Contain highly conserved proteins do multiple sequence alignment by muscle.</li>
<li>
<p>concatenate: Contain the concatenate highly conserved proteins result and select conserved blocks data.</p>
<ul>
<li>concatenate.fasta: Concatenate highly conserved proteins result.</li>
<li>concatenate.fasta-gb1: Select conserved blocks result (by Gblocks).</li>
<li>concatenate.fasta-gb1.htm: Select conserved blocks result view by html format.</li>
<li>concatenate.fasta-gb1.phy: Convert FASTA format to PHYLIP format</li>
</ul>
</li>
</ul>
</li>
</ul>
<h4 id="auto-build-phylogenetic-tree-by-16s-ssu-rna"><strong>Auto build phylogenetic tree by 16s SSU RNA:</strong></h4>
<pre><code class="bash">$ physpe autobuild -i organism_example_list.txt --srna
Loading organisms names success.....

The result are store in:Outdata

Now loading data and constructing species phylogenetic tree......
2016-08-18 15:29:31,528 16s DB INFO: Read organisms names success
2016-08-18 15:29:38,166 16s DB INFO: Retrieve organism 'aca' 16s SSU RNA sequences data success
2016-08-18 15:29:40,236 16s DB INFO: Retrieve organism 'ace' 16s SSU RNA sequences data success
2016-08-18 15:29:42,356 16s DB INFO: Retrieve organism 'acl' 16s SSU RNA sequences data success
2016-08-18 15:29:44,615 16s DB INFO: Retrieve organism 'acn' 16s SSU RNA sequences data success
2016-08-18 15:29:46,868 16s DB INFO: Retrieve organism 'aco' 16s SSU RNA sequences data success
2016-08-18 15:29:49,119 16s DB INFO: Retrieve organism 'acp' 16s SSU RNA sequences data success
2016-08-18 15:29:51,392 16s DB INFO: Retrieve organism 'adg' 16s SSU RNA sequences data success

</code></pre>

<p>When auto reconstruct phylogenetic tree by <code>srna</code> method was completed, you will get file layout like this:</p>
<pre><code>log.log
Outdata/
        RAxML_bestTree.T1
        RAxML_bipartitions.T1
        RAxML_bipartitionsBranchLabels.T1
        RAxML_bootstrap.T1
        RAxML_info.T1
   temp/
        16srnadata20160818145209/
                                 16srandata.fasta

        16srna_alignment20160818145236/
                                  16srandata.fasta
                                  16srandata.fasta-gb1
                                  16srandata.fasta-gb1.htm
                                  16srandata.fasta-gb1.phy
</code></pre>

<ul>
<li>log.log: The log information of physpe.</li>
<li>
<p>Outdata: Contain phylogenetic tree output result. </p>
<ul>
<li>RAxML_bestTree.T1: Reconstruct phylogenetic tree by RAxML, it's best ML search tree. </li>
<li>RAxML_bipartitions.T1: Bipartition tree by RAxML.</li>
<li>RAxML_bipartitionsBranchLabels.T1: Bipartition tree by constructed by RAxML with branch length lables.</li>
<li>RAxML_bootstrap.T1: Bootstrap result by RAxML</li>
<li>RAxML_info.T1: The info of run RAxMl.</li>
</ul>
</li>
<li>
<p>temp: The temp data by physpe, <code>it's very important to user to check the key steps</code>.</p>
<ul>
<li>
<p>16srnadata: Contain 16s SSU RNA data retrieved form SILVA database.</p>
<ul>
<li>16srnadata.fasta: FASTA format data contain input organism 16s SSU RNA data.</li>
</ul>
</li>
<li>
<p>16srna_alignment: Contain the concatenate highly conserved proteins result and select conserved blocks data.</p>
<ul>
<li>16srandata.fasta: Contain 16s SSU RNA data with multiple sequence alignment already.</li>
<li>16srandata.fasta-gb1: Select conserved blocks result (by Gblocks).</li>
<li>16srandata.fasta-gb1.htm: Select conserved blocks result view by html format.</li>
<li>16srandata.fasta-gb1.phy: Convert FASTA format to PHYLIP format</li>
</ul>
</li>
</ul>
</li>
</ul>
<h3 id="advance-options">Advance options</h3>
<p>User enable choice more detail options with Physpe call software, detail advance options input
<code>must be enclosed in single quotes</code>.</p>
<p>The follow is to use RAxML advance options example:</p>
<pre><code class="bash">$ physpe -i organism_example_list.txt --raxml '-f a -m GTRGAMMA  -p 12345 -x 12345 -# 100 -n T1'
</code></pre>

<p><strong>--muscle</strong></p>
<p>Multiple sequence alignment by muscle. The default aligned software is Muscle.</p>
<p><strong>--muscle_p</strong></p>
<p>Set multiple sequence alignment parameters. The default is <code>-maxiter 100</code>. More options about muslce please to see <a href="http://www.drive5.com/muscle/manual/options.html">MUSCLE Manual</a></p>
<p>The default option:</p>
<table>
<thead>
<tr>
<th align="left">option</th>
<th align="left">description</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">-maxiter</td>
<td align="left">maximum number of iterations to run is set 100.</td>
</tr>
</tbody>
</table>
<p><strong>--clustalw</strong></p>
<pre><code>Multiple sequense alignment by clustalw2.
</code></pre>
<p><strong>--clustalw_p</strong></p>
<p>Set more detail clustalw2 parameters. Here use clustalw default parameters. More options about clustalw please to see <a href="http://www.clustal.org/download/clustalw_help.txt">Clustalw Help</a>.</p>
<p><strong>--gblocks</strong></p>
<p>Set Gblocks parameters. The default is <code>-t=p -e=-gb1</code>. More options about Gblocks please to see <a href="http://molevol.cmima.csic.es/castresana/Gblocks/Gblocks_documentation.html">Gblocks documentation</a>.</p>
<p>The default option:</p>
<table>
<thead>
<tr>
<th align="left">option</th>
<th align="left">description</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">-t</td>
<td align="left">Choice type of sequence. The physpe default set is protein.</td>
</tr>
<tr>
<td align="left">-e</td>
<td align="left">Eneric file extensionc. physep set default is -gbl1.</td>
</tr>
</tbody>
</table>
<p><strong>--ranxml</strong></p>
<pre><code>Reconstruct phylogenetic tree by RAxML. The default build tree software is RAxML.
</code></pre>
<p><strong>--raxml_p</strong></p>
<p>Set reconstruct phylogenetic tree arguments with RAxML. The default is <code>-f a -m PROTGAMMAJTTX  -p 12345 -x 12345 -# 100 -n T1</code>. More options about RAxMl please to see <a href="http://sco.h-its.org/exelixis/resource/download/NewManual.pdf">RAxML Manual</a>.</p>
<p>The default option:</p>
<table>
<thead>
<tr>
<th align="left">option</th>
<th align="left">description</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">-f</td>
<td align="left">select algorithm. The physpe default set is <code>a</code>, rapid Bootstrap analysis and search for best­scoring ML tree in one program run.</td>
</tr>
<tr>
<td align="left">-m</td>
<td align="left">Model of Binary (Morphological), Nucleotide, Multi­State, or Amino Acid Substitution. The physpe default set is PROTGAMMAJTTX.</td>
</tr>
<tr>
<td align="left">-p</td>
<td align="left">Specify a random number seed for the parsimony inferences. The physep default set is 12345.</td>
</tr>
<tr>
<td align="left">-x</td>
<td align="left">Specify an integer number (random seed) and turn on rapid bootstrapping. The physpe default set is 12345.</td>
</tr>
<tr>
<td align="left">-N</td>
<td align="left">The same with -# specify the number of alternative runs on distinct starting trees. The physpe default set is 100.</td>
</tr>
</tbody>
</table>
<p><strong>--fasttree</strong></p>
<p>Reconstruct phylogenetic tree by FastTree.</p>
<p><strong>--fasttree_p</strong></p>
<p>Set more detail RAxML parameters.More options about clustalw please to see <a href="http://www.microbesonline.org/fasttree/">FastTree</a>.</p>
<h2 id="build">build</h2>
<p>User can build tree by own 16s rna data or highly conserved proteins.</p>
<p>Use <strong>build</strong> in command line to reconstruct phylogenetic tree:</p>
<ul>
<li>build phylogenetic tree by highly conserved proteins</li>
</ul>
<pre><code class="bash">$ physpe build -i example_hcp -o output --hcp
</code></pre>

<ul>
<li>build phylogenetic tree by 16s ssu rna data</li>
</ul>
<pre><code class="bash">$ physpe build -i example_16s_ssurna.fasta -o output --sran
</code></pre>

<h3 id="build-options">build options</h3>
<table>
<thead>
<tr>
<th align="left">option</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">-h</td>
<td align="left">Print help message and exits.</td>
</tr>
<tr>
<td align="left">-i</td>
<td align="left">Input a txt file contain the a abbreviation species names are same with KEGG species abbreviation.</td>
</tr>
<tr>
<td align="left">-o</td>
<td align="left">A directory include output data (reconstruct tree files). The default output data name is Outdata.</td>
</tr>
<tr>
<td align="left">-t</td>
<td align="left">Specify the number of processing threads (CPUs) to use for Physpe to reconstruct phylogenetic tree. The default is 1.</td>
</tr>
<tr>
<td align="left">--hcp</td>
<td align="left">The hcp (highly conserved protein) mode is use highly conserved proteins to reconstruct phylogenetic tree. The default mode is hcp.</td>
</tr>
<tr>
<td align="left">--srna</td>
<td align="left">The 16srna (16 SSU RNA) mode is use 16s SSU RNA data to reconstruct phylogenetic tree.</td>
</tr>
</tbody>
</table>
<h3 id="example_1">example</h3>
<h4 id="build-phylogenetic-tree-by-highly-conserved-proteins"><strong>Build phylogenetic tree by highly conserved proteins</strong></h4>
<p>When use build to reconstruct phylogenetic tree you should prepare a directory contain highly conserved proteins such as <a href="../example/example_build_srna.fasta">example_build_hcp</a>. In this example
contain ten highly conserved proteins p1~p10, in each highly conserved proteins contain 10 organism.</p>
<p>Download <a href="../example/example_build_srna.fasta">example_build_hcp</a> you can use the tar command to unpack:</p>
<pre><code class="bash">$ tar -zxvf example_build_hcp.tar.gz                                                          
example_build_hcp/                                                                                                   
example_build_hcp/p1.fasta
example_build_hcp/p2.fasta
example_build_hcp/p3.fasta
example_build_hcp/p4.fasta
example_build_hcp/p5.fasta
example_build_hcp/p6.fasta
example_build_hcp/p7.fasta
example_build_hcp/p8.fasta
example_build_hcp/p9.fasta
example_build_hcp/p10.fasta
</code></pre>

<p>Check each highly conserved proteins in each files:</p>
<pre><code class="bash">$ cd  example_build_hcp/
$ cat p1.fasta 
&gt;aeh
MARLTKRQKAIREKIDPAQQYPVAEALGLLRELPGAKFTESVEVAVNLGVDPRKSDQIVR
GSTVLPNGTGKTVRVAVFAQGDAAEAAKEAGADIVGMDDLAEQVKGGNLDFDVVVAAPDA
MGVVGRLGPILGPRGLMPNPKVGTVTPDVAGAVKNAKAGQVRYRTDKGGIIHCAIGKVDF
EVEALQQNLQALITDLQKLKPANSKGVYLKKVAVSTTMGPGLAVDLASLET
&gt;adk
MAKLTKKQKAQQGKVDSTKLYPFAEAVALVKEAATAKFDESIDVAVQLGVDAKKSDQVVR
GAVVLPNGTGKTTRVAVFAQGAKAEEAKAAGADVVGMDDLAAQVKAGDMPFDVVIAAPDA
MRVVGTLGQILGPRGLMPNPKVGTVTPDVATAVKNAKAGQVQFRVDKAGIVHTTIGRRSF
ADDKLQGNLAALIEALNKAKPATSKGVYLRKVAVSSTMGVGVRVDTQSIAA
&gt;acp
MAHVAKKYKAAAEKVDRTKRYKLDEAMSLVKQTATKKFDETVDASINLGVDPKHADQVVR
GAVVLPHGMGKTVRLAVFAKGDKAKEAQEAGADIVGAEDLAEKIQGGFMDFDKLIATPDM
MGVVGRLGKILGPRGLMPNPKVGTVTMDLARAVKEQKAGKVEFRVEKAGIVHVPFGKASF
DPDKLKANFSAIMEVIYKAKPQTAKGVYVKNVTLSTTMGPGIKVDLAELAAQHA
&gt;acn
MSGDGSSYSAEEGIRELLQSAKAKFRESVDVAIKLSVADSKSGESVRGAVVLPKGLGREV
RVAVFAKGEHAKHASDAGADVVGDEELIEEIKKGRKLDVDWCIATPDFMPQISAIAKILG
PRGLMPNPKFGTVTLELAKMVGVIKSGQVKFKSDRYGIVHVKIGDVSFTPEDLLENFNAV
VVAVQNLKPATIKGSYVRGVFVNSTMGRSFRIAGIG
&gt;adg
MPKHGKKYLEAKKQVDRTKLYDPYEALELVKRLASAKFDETVEVAVRLGVDPRHADQQVR
GAVVLPHGTGKTRRVLVFARGEKAKEAEAAGADYVGAEDLIARIQGGWLDFDVAIATPDM
MAMVGRIGRILGPRGLMPNPKTGTVTFDVAQAVAEAKAGRVEYRTDKAGIVHAPIGKVSF
EVEKLVENLKALVDALVRAKPPAAKGQYLRSITVSSTMGPGVKVNPAKLLAS
&gt;acl
MKRGKKYLEAVKLYDKSVAYTGLEAVELAKKTSVAKFDATVEVAFRLNVDPRKADQNLRG
AISLPHGTGKTVRVVVIAKPEKAKEALAAGALEAGDVELIDKIGKGWFDFDVMVATPDMM
AQLGKLGRVLGPKGLMPNPKTGTVTLDVAKAVEEIKAGKIEYRTDKVGNIHAPIGKVSFD
SNKLHENMLAIYNQLVRIKPATVKGTYIKKIALSTTMGPGIMVEENNIKK
&gt;ace
MKRGKKYRAAAQLVDRTKLYSPLEAMRLAKQTNTMRVPATVEVAMRLGVDPRKADQMVRG
TVNLPHGTGKTPRVLVFATAERAEEARAAGADYVGADELIEQVANGFLDFDAVVATPDLM
GKVGRLGRILGPRGLMPNPKTGTVTNDVAKAVADIKSGKIEFRVDRQANLHLVIGKTDFT
EQQLVENYAAALDEVLRLKPPTAKGRYLKKVTISTTMGPGIPVDPNRVRNLLAEETAAA
&gt;aeq
MTKHGKKYVEAEKQIPAEPVSPLAAMKLLKEISVANFDETVTGDFRLGIDTRQADQQLRG
TVSLPNGSGKTVRVAVFAEGAAAQAAEEAGADIVGTDELMQQIQAGEFNFDAAVATPDQM
GKVGRLGKILGPRGLMPNPKLGTVTNDVAKAIKELKGGRVEYRADRYGIAHVVLGKVSFT
PEQLAENYGAVYDEILRMKPAAAKGKYVKSITVSGTMTPGVSVDSSVTRAYTESAE
&gt;aca
MSKKVSKNVAKARAAVEPRPYTLQDAVPLLQQVKFAKFDETVDLTMRLGVDPRHADQMVR
GTVVLPHGLGKTKKVAVITTGDRQKEAEAAGAEIVGGEELVEKIQKESWTDFDALIATPD
MMRSVGRLGKVLGPRGLMPNPKTGTVTNDVAAAVKEIKAGKIEYRTDKTALVHVPVGKLS
FPAEKLIDNAMTVITSVVRAKPSAAKGKYIKGITLSSTMGPGIPLDGSVADAAAKA
&gt;aco
MAKKSKRYSEIAAKVDSTKLYGLREAVDLYKEVATAKFDESLEVHIRLGVDPRHADQQVR
GTIVLPHGTGITKRVLVLAVGEKVKEAEDAGADIVGGDDLIQKISTGWLDFDAVIATPDM
MKSVGRLGKILGPRGLMPSAKAGTVTFDVADAIKEIKAGRVEFRVDKTAIIHNMVGKKSF
EAEKLFENLKVLYRAILKARPASAKGTYVRSFYIAPTMGVGIKIDPVAASKEVAEA
</code></pre>

<p>Reconstruct phylogenetic tree ues example are by follow command:</p>
<pre><code class="bash">physpe build -i example_build_hcp -o build_hcp_tree --hcp
</code></pre>

<p>Few seconds reconstruct phylogenetic tree completed and tree file are store in <code>build_hcp_tree</code> directory</p>
<p><code>NOTE</code>:</p>
<ul>
<li>Prepare how many highly proteins which user decide, we recommend not less than 10 highly conserved proteins.</li>
<li>Each protein must have the same organisms.</li>
</ul>
<h4 id="build-phylogenetic-tree-by-16s-rna"><strong>Build phylogenetic tree by 16s RNA</strong></h4>
<p>Use build command to construct phylogenetic tree by 16s RNA sequence, user should prepare a FASTA format file
contain each organisms 16s RNA sequence as <a href="../example/example_combine_tree.tar.gz">example_build_srna</a>. In this example contain ten organism 16s SSU RNA sequence.</p>
<p>Download <a href="../example/example_combine_tree.tar.gz">example_build_srna</a> and check the organisms names:</p>
<pre><code class="bash">$ grep '&gt;' example_build_srna.fasta 
&gt;aca
&gt;ace
&gt;acl
&gt;acn
&gt;aco
&gt;acp
&gt;adg
&gt;adk
&gt;aeh
&gt;aeq
</code></pre>

<p>Reconstruct phylogenetic tree ues example are by follow command:</p>
<pre><code class="bash">physpe build -i example_build_srna.fasta -o build_srna_tree --srna
</code></pre>

<p>Few seconds reconstruct phylogenetic tree completed and tree file are store in <code>build_srn_tree</code> directory</p>
<h3 id="advance-options_1">Advance options</h3>
<p>User enable choice more detail options with Physpe call software, detail advance options input
<code>must be enclosed in single quotes</code>.</p>
<p>The follow is to use RAxML advance options example:</p>
<pre><code class="bash">$ physpe -i example_chp  --raxml '-f a -m GTRGAMMA  -p 12345 -x 12345 -# 100 -n T1' --hcp
</code></pre>

<p><strong>--muscle</strong></p>
<pre><code>Multiple sequence alignment by muscle. The default aligned software is Muscle.
</code></pre>
<p><strong>--muscle_p</strong></p>
<p>Set multiple sequence alignment parameters. The default is <code>-maxiter 100</code>. More options about muslce please to see <a href="http://www.drive5.com/muscle/manual/options.html">MUSCLE Manual</a></p>
<p>The default option:</p>
<table>
<thead>
<tr>
<th align="left">option</th>
<th align="left">description</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">-maxiter</td>
<td align="left">maximum number of iterations to run is set 100.</td>
</tr>
</tbody>
</table>
<p><strong>--clustalw</strong></p>
<p>Multiple sequense alignment by clustalw2.</p>
<p><strong>--clustalw_p</strong></p>
<p>Set more detail clustalw2 parameters. Here use clustalw default parameters. More options about clustalw please to see <a href="http://www.clustal.org/download/clustalw_help.txt">Clustalw Help</a>.</p>
<p><strong>--gblocks</strong></p>
<p>Set Gblocks parameters. The default is <code>-t=p -e=-gb1</code>. More options about Gblocks please to see <a href="http://molevol.cmima.csic.es/castresana/Gblocks/Gblocks_documentation.html">Gblocks documentation</a>.</p>
<p>The default option:</p>
<table>
<thead>
<tr>
<th align="left">option</th>
<th align="left">description</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">-t</td>
<td align="left">Choice type of sequence. The physpe default set is protein.</td>
</tr>
<tr>
<td align="left">-e</td>
<td align="left">Eneric file extensionc. physep set default is -gbl1.</td>
</tr>
</tbody>
</table>
<p><strong>--ranxml</strong></p>
<p>Reconstruct phylogenetic tree by RAxML. The default build tree software is RAxML.</p>
<p><strong>--raxml_p</strong></p>
<p>Set reconstruct phylogenetic tree arguments with RAxML. The default is <code>-f a -m PROTGAMMAJTTX  -p 12345 -x 12345 -# 100 -n T1</code>. More options about RAxMl please to see <a href="http://sco.h-its.org/exelixis/resource/download/NewManual.pdf">RAxML Manual</a>.</p>
<p>The default option:</p>
<table>
<thead>
<tr>
<th align="left">option</th>
<th align="left">description</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">-f</td>
<td align="left">select algorithm. The physpe default set is <code>a</code>, rapid Bootstrap analysis and search for best­scoring ML tree in one program run.</td>
</tr>
<tr>
<td align="left">-m</td>
<td align="left">Model of Binary (Morphological), Nucleotide, Multi­State, or Amino Acid Substitution. The physpe default set is PROTGAMMAJTTX.</td>
</tr>
<tr>
<td align="left">-p</td>
<td align="left">Specify a random number seed for the parsimony inferences. The physep default set is 12345.</td>
</tr>
<tr>
<td align="left">-x</td>
<td align="left">Specify an integer number (random seed) and turn on rapid bootstrapping. The physpe default set is 12345.</td>
</tr>
<tr>
<td align="left">-N</td>
<td align="left">The same with -# specify the number of alternative runs on distinct starting trees. The physpe default set is 100.</td>
</tr>
</tbody>
</table>
<p><strong>--fasttree</strong></p>
<p>Reconstruct phylogenetic tree by FastTree.</p>
<p><strong>--fasttree_p</strong></p>
<p>Set more detail RAxML parameters.More options about clustalw please to see <a href="http://www.microbesonline.org/fasttree/">FastTree</a>.</p>
<h2 id="combine">combine</h2>
<p>User should prepare a combine tree file by Combine command to combine tree files.</p>
<p>In Linux you can easy combine more tree to a tree file, for example:</p>
<pre><code class="bash">$ cat tree1.tree tree2.tree &gt; combieTree.tree

</code></pre>

<p>Use <strong>combine</strong> in command line like this:</p>
<pre><code class="bash">$ physpe -i organism_example_list.txt [options]*
</code></pre>

<h3 id="combine-options">Combine options</h3>
<table>
<thead>
<tr>
<th>option</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>-h</td>
<td>Print help message and exits.</td>
</tr>
<tr>
<td>-i</td>
<td>Input a txt file contain the a abbreviation species names are same with KEGG species abbreviation.</td>
</tr>
<tr>
<td>-o</td>
<td>A directory contain combine tree file. The default output data name is combinetree.</td>
</tr>
</tbody>
</table>
<h3 id="example_2">example</h3>
<p>User can use combine command to combine tree from different method method, in our example combine tree from two way reconstruct
The <code>tree1.tree</code> reconstruct by highly conserved proteins, the <code>tree2.tree</code> reconstruct by 16s SSU RNA data, exapmel data
<a href="../example/example_combine_tree.tar.gz">example_combine_tree.tar.gz</a></p>
<p>Download <a href="../example/example_combine_tree.tar.gz">example_combine_tree.tar.gz</a> and unpack:</p>
<pre><code class="bash">$ tar -zxvf example_combine_tree.tar.gz 
example_combine_tree/
example_combine_tree/tree2.tree
example_combine_tree/tree1.tree
</code></pre>

<p>Combine tree1.tree and tree2.tree to a tree file:</p>
<pre><code class="bash">$ conbine tree1.tree tree2.tree &gt; combine.tree
</code></pre>

<p>Combine tree command as like this:</p>
<pre><code class="bash">physpe combine -i combine.tree -o combineTree
</code></pre>

<p>In the combineTree directory contain two files:</p>
<pre><code>combine/
        RAxML_info.T1
        RAxML_MajorityRuleConsensusTree.T1
</code></pre>

<ul>
<li>RAxML_info.T1: The info of RAxML</li>
<li>RAxML_MajorityRuleConsensusTree.T1: The combine tree by Majority Rule method.</li>
</ul>
<h2 id="iview">iview</h2>
<p>Annotating tree by iTol use iview module. User can use iview command to color tree range by kingdom, phylum, class or order.
the range annotation files can used in <a href="http://itol.embl.de/">iTol</a>, iTol is a very popular online tool for the display, 
annotation and management of phylogenetic trees. More detail with of iTol in <a href="http://itol.embl.de/help.cgi">iTol help</a> </p>
<p>Use <strong>iview</strong> in command line like this:</p>
<pre><code class="bash">$ physpe iview -i organism_example_list.txt -range phylum
</code></pre>

<h3 id="iview-options">iview options</h3>
<table>
<thead>
<tr>
<th align="left">option</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">-h</td>
<td align="left">Print help message and exits.</td>
</tr>
<tr>
<td align="left">-i</td>
<td align="left">Input a txt file contain the a abbreviation species names are same with KEGG species abbreviation.</td>
</tr>
<tr>
<td align="left">-o</td>
<td align="left">A directory contain range text file. The directory name is iview.</td>
</tr>
<tr>
<td align="left">-a</td>
<td align="left">Colored ranges by user assign, user can choice from kingdom phylum class and order.</td>
</tr>
<tr>
<td align="left">-r/--range</td>
<td align="left">Annotating ranges by kingdom, phylum, class or order. The default is phylum.</td>
</tr>
<tr>
<td align="left">-l/--labels</td>
<td align="left">Change labels from abbreviation names to full names.</td>
</tr>
</tbody>
</table>
<h3 id="example_3">example</h3>
<p>iview Annotating tree by kingdom, phylum, class or order example, download example <a href="../example/organism_example_list.txt">organism_example_list.txt</a></p>
<h4 id="annotating-tree-by-kingdom">Annotating tree by kingdom</h4>
<pre><code class="bash">$ physpe iview -i organism_example_list.txt --range kingdom
Color range by kingdom was complete.
</code></pre>

<p>The color range type data are store in iview directory:</p>
<pre><code class="bash">$ cd iview
$ cat range_color_by_kingdom.txt 
TREE_COLORS
SEPARATOR TAB
DATA
aca     range   #BEBF5A Prokaryotes
ace     range   #BEBF5A Prokaryotes
acl     range   #BEBF5A Prokaryotes
acn     range   #BEBF5A Prokaryotes
aco     range   #BEBF5A Prokaryotes
acp     range   #BEBF5A Prokaryotes
adg     range   #BEBF5A Prokaryotes
adk     range   #BEBF5A Prokaryotes
aeh     range   #BEBF5A Prokaryotes
aeq     range   #BEBF5A Prokaryotes
</code></pre>

<h4 id="annotating-tree-by-phylum">Annotating tree by phylum</h4>
<pre><code class="bash">$ physpe iview -i organism_example_list.txt --range phylum
Color range by phylum was complete.
</code></pre>

<p>The color range type data are store in iview directory:</p>
<pre><code class="bash">$ cd iview
$ cat range_color_by_phylum.txt 
TREE_COLORS
SEPARATOR TAB
DATA
aca     range   #865142 Bacteria
ace     range   #865142 Bacteria
acl     range   #865142 Bacteria
acn     range   #865142 Bacteria
aco     range   #865142 Bacteria
acp     range   #865142 Bacteria
adg     range   #865142 Bacteria
adk     range   #865142 Bacteria
aeh     range   #865142 Bacteria
aeq     range   #865142 Bacteria
</code></pre>

<h4 id="annotating-tree-by-class">Annotating tree by class</h4>
<pre><code class="bash">$ physpe iview -i organism_example_list.txt --range class
Color range by class was complete.
</code></pre>

<p>The color range type data are store in iview directory:</p>
<pre><code class="bash">$ cd iview
$ cat range_color_by_class.txt 
TREE_COLORS
SEPARATOR TAB
DATA
aca     range   #9AB7F3 Acidobacteria
ace     range   #99D1DB Actinobacteria
acl     range   #A5E58D Tenericutes
acn     range   #94F1C1 Alphaproteobacteria
aco     range   #D67A21 Synergistetes
acp     range   #DD9284 Deltaproteobacteria
adg     range   #3E70B8 Firmicutes - Clostridia
adk     range   #DDC8B7 Betaproteobacteria
aeh     range   #72E137 Gammaproteobacteria - Others
aeq     range   #99D1DB Actinobacteria
</code></pre>

<h4 id="annotating-tree-by-order">Annotating tree by order</h4>
<pre><code class="bash">$ physpe iview -i organism_example_list.txt --range order
Color range by order was complete.
</code></pre>

<p>The color range type data are store in iview directory:</p>
<pre><code class="bash">$ cd iview
$ cat range_color_by_order.txt 
TREE_COLORS
SEPARATOR TAB
DATA
aca     range   #AA8761 Acidobacterium
ace     range   #8770BC Acidothermus
acl     range   #3BD26B Acholeplasma
acn     range   #D1B487 Anaplasma
aco     range   #D96D21 Aminobacterium
acp     range   #AC4E16 Anaeromyxobacter
adg     range   #287AD8 Ammonifex
adk     range   #C8184E Alicycliphilus
aeh     range   #57A569 Alkalilimnicola
aeq     range   #F1A2B7 Adlercreutzia
</code></pre>

<h2 id="check">check</h2>
<p>Use check module  check input organisms match in kegg database or 16s database.</p>
<pre><code class="bash">$ physpe check -i organism_example_list.txt -out check --ehcp
</code></pre>

<h3 id="check-options">check options</h3>
<table>
<thead>
<tr>
<th align="left">option</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">-h</td>
<td align="left">Print help message and exits.</td>
</tr>
<tr>
<td align="left">-i</td>
<td align="left">Input a txt file contain the a abbreviation species names are same with KEGG species abbreviation.</td>
</tr>
<tr>
<td align="left">-o</td>
<td align="left">A directory contain check result. The directory name is check.</td>
</tr>
<tr>
<td align="left">--ehcp</td>
<td align="left">check input organisms prepare for extend autobuild tree module.</td>
</tr>
</tbody>
</table>
<h3 id="example_4">example</h3>
<h4 id="check-organism-for-extend-autobuild">check organism for extend autobuild</h4>
<p>When user use autobuild to reconstruct phylogenetic tree with <code>--ehcp</code> method, you should prepare highly conserved proteins as extend proteins
for auto build tree. You can determine what proteins to be prepared by the check command. Download example for check <a href="../example/organism_example_list.txt">organism_example_list.txt</a></p>
<p>Use check command like follow: </p>
<pre><code class="bash">$ physpe check -i organism_example_list.txt --ehcp

'Ribosomal protein L1' ----------------------------------&gt; p1.fasta

'DNA-directed RNA polymerase subunit alpha' ----------------------------------&gt; p2.fasta

'Leucyl-tRNA synthetase' ----------------------------------&gt; p3.fasta

'Metal-dependent proteases with chaperone activity' ----------------------------------&gt; p4.fasta

'Phenylalanine-tRNA synthethase alpha subunit' ----------------------------------&gt; p5.fasta

'Predicted GTPase probable translation factor' ----------------------------------&gt; p6.fasta

'Ribosomal protein L11' ----------------------------------&gt; p7.fasta

'Ribosomal protein L13' ----------------------------------&gt; p8.fasta

'Ribosomal protein L14' ----------------------------------&gt; p9.fasta

'Ribosomal protein L22' ----------------------------------&gt; p10.fasta

'Ribosomal protein L3' ----------------------------------&gt; p11.fasta

'Ribosomal protein L5' ----------------------------------&gt; p12.fasta

'Ribosomal protein S11' ----------------------------------&gt; p13.fasta

'Ribosomal protein S17' ----------------------------------&gt; p14.fasta

'Ribosomal protein S2' ----------------------------------&gt; p15.fasta

'Ribosomal protein S3' ----------------------------------&gt; p16.fasta

'Ribosomal protein S4' ----------------------------------&gt; p17.fasta

'Ribosomal protein S5' ----------------------------------&gt; p18.fasta

'Ribosomal protein S7' ----------------------------------&gt; p19.fasta

'Ribosomal protein S8' ----------------------------------&gt; p20.fasta

'Ribosomal protein S9' ----------------------------------&gt; p21.fasta

'Seryl-tRNA synthetase' ----------------------------------&gt; p22.fasta

'Arginyl-tRNA synthetase' ----------------------------------&gt; p23.fasta

'DNA-directed RNA polymerase beta subunit' ----------------------------------&gt; p24.fasta

'Ribosomal protein S13' ----------------------------------&gt; p25.fasta

Check extend highly conserved protein is completed.

</code></pre>

<p>This check result write to default directory name is <code>check</code></p>
<pre><code class="bash">$ cd check
$ cat physpe_echp_extend.txt 
'Ribosomal protein L1' ----------------------------------&gt; p1.fasta
'DNA-directed RNA polymerase subunit alpha' ----------------------------------&gt; p2.fasta
'Leucyl-tRNA synthetase' ----------------------------------&gt; p3.fasta
'Metal-dependent proteases with chaperone activity' ----------------------------------&gt; p4.fasta
'Phenylalanine-tRNA synthethase alpha subunit' ----------------------------------&gt; p5.fasta
'Predicted GTPase probable translation factor' ----------------------------------&gt; p6.fasta
'Ribosomal protein L11' ----------------------------------&gt; p7.fasta
'Ribosomal protein L13' ----------------------------------&gt; p8.fasta
'Ribosomal protein L14' ----------------------------------&gt; p9.fasta
'Ribosomal protein L22' ----------------------------------&gt; p10.fasta
'Ribosomal protein L3' ----------------------------------&gt; p11.fasta
'Ribosomal protein L5' ----------------------------------&gt; p12.fasta
'Ribosomal protein S11' ----------------------------------&gt; p13.fasta
'Ribosomal protein S17' ----------------------------------&gt; p14.fasta
'Ribosomal protein S2' ----------------------------------&gt; p15.fasta
'Ribosomal protein S3' ----------------------------------&gt; p16.fasta
'Ribosomal protein S4' ----------------------------------&gt; p17.fasta
'Ribosomal protein S5' ----------------------------------&gt; p18.fasta
'Ribosomal protein S7' ----------------------------------&gt; p19.fasta
'Ribosomal protein S8' ----------------------------------&gt; p20.fasta
'Ribosomal protein S9' ----------------------------------&gt; p21.fasta
'Seryl-tRNA synthetase' ----------------------------------&gt; p22.fasta
'Arginyl-tRNA synthetase' ----------------------------------&gt; p23.fasta
'DNA-directed RNA polymerase beta subunit' ----------------------------------&gt; p24.fasta
'Ribosomal protein S13' ----------------------------------&gt; p25.fasta
</code></pre>

<p>In <code>physpe_echp_extend.txt</code> file you can check the highly conserved protein, which you should prepare for extend hcp method to reconstruct phylogenetic tree.</p></div>
            
        </div>

        <footer class="col-md-12">
            <hr>
            
            <p>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</p>
        </footer>

        <script src="../js/jquery-1.10.2.min.js"></script>
        <script src="../js/bootstrap-3.0.3.min.js"></script>
        <script src="../js/highlight.pack.js"></script>
        <script>var base_url = '..';</script>
        <script data-main="../mkdocs/js/search.js" src="../mkdocs/js/require.js"></script>
        <script src="../js/base.js"></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
            <div class="modal-dialog">
                <div class="modal-content">
                    <div class="modal-header">
                        <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                        <h4 class="modal-title" id="exampleModalLabel">Search</h4>
                    </div>
                    <div class="modal-body">
                        <p>
                            From here you can search these documents. Enter
                            your search terms below.
                        </p>
                        <form role="form">
                            <div class="form-group">
                                <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
                            </div>
                        </form>
                        <div id="mkdocs-search-results"></div>
                    </div>
                    <div class="modal-footer">
                    </div>
                </div>
            </div>
        </div>

    </body>
</html>
